server/src/main/java/com/scalar/db/server/ScalarDbServer.java

```diff
@@ -120,12 +123,20 @@ public void addShutdownHook() {
         .addShutdownHook(
             new Thread(
                 () -> {
-                  logger.info("Signal received. Shutting down the server ...");
-                  shutdown(MAX_WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS);
-                  logger.info("The server shut down.");
+                  logger.info("Signal received. Decommissioning ...");
+                  decommission();
```

Comment:
Before shutting down ScalarDB Server, change the healthService response to `NOT_SERVING` and sleep for some time. This way, we can wait for no new requests to come in.